internal/pkg/service/templates/dependencies/reqproject_test.go

```diff
@@ -54,7 +54,7 @@ func TestProjectRequestScope_TemplateRepository_Cached(t *testing.T) {
 	// FS contains template1, but doesn't contain template2
 	assert.NoError(t, err)
 	assert.True(t, repo1.Fs().Exists(ctx, "template1"))
-	assert.False(t, repo1.Fs().Exists(ctx, "template2"))
+	assert.False(t, repo1.Fs().Exists(ctx, "template4"))
 
 	// Update repository -> no change
```

Comment:
`template4` is not in any commit, so I would say that you could just delete that assert at all :smiley: 

The meaning of the test is that `template2` is not there and after `git reset --hard' it is there.

So I suggest to add `runGitCommand(t, tmpDir, "reset", "--hard", "<hash>")` to the beginning of the test, to keep original logic.